engine-nextgen/src/main/kotlin/graphql/nadel/validation/util/NadelGetReachableTypes.kt

```diff
@@ -0,0 +1,245 @@
+package graphql.nadel.validation.util
+
+import graphql.nadel.Service
+import graphql.nadel.enginekt.util.unwrapAll
+import graphql.nadel.validation.util.NadelCombinedTypeUtil.getFieldsThatServiceContributed
+import graphql.nadel.validation.util.NadelCombinedTypeUtil.isCombinedType
+import graphql.nadel.validation.util.NadelSchemaUtil.hasHydration
+import graphql.schema.GraphQLArgument
+import graphql.schema.GraphQLCompositeType
+import graphql.schema.GraphQLDirective
+import graphql.schema.GraphQLDirectiveContainer
+import graphql.schema.GraphQLEnumType
+import graphql.schema.GraphQLFieldDefinition
+import graphql.schema.GraphQLFieldsContainer
+import graphql.schema.GraphQLInputFieldsContainer
+import graphql.schema.GraphQLInputObjectField
+import graphql.schema.GraphQLInputObjectType
+import graphql.schema.GraphQLInterfaceType
+import graphql.schema.GraphQLList
+import graphql.schema.GraphQLModifiedType
+import graphql.schema.GraphQLNamedSchemaElement
+import graphql.schema.GraphQLNonNull
+import graphql.schema.GraphQLNullableType
+import graphql.schema.GraphQLObjectType
+import graphql.schema.GraphQLOutputType
+import graphql.schema.GraphQLScalarType
+import graphql.schema.GraphQLSchema
+import graphql.schema.GraphQLSchemaElement
+import graphql.schema.GraphQLTypeReference
+import graphql.schema.GraphQLTypeVisitorStub
+import graphql.schema.GraphQLUnionType
+import graphql.schema.GraphQLUnmodifiedType
+import graphql.schema.SchemaTraverser
+import graphql.util.TraversalControl
+import graphql.util.TraversalControl.ABORT
+import graphql.util.TraversalControl.CONTINUE
+import graphql.util.TraverserContext
+
+internal fun getReachableTypeNames(
```

Comment:
So what this does, is it takes in all the types defined into the overall schema i.e. `Service.getDefinitionRegistry`

Then it starts looking at all the possible reachable types via `SchemaTraverser`. Then it collects these into a `Set` that it returns.

Note, we still ignore hydration output types, and handle combined types (i.e. `Query` and namespaced operation types) when they're referenced.